### DISCUSSION TITLE
ace editor html beautify in kitchen demo

### DISCUSSION THREAD

**Discussion by ldijkman** [ajaxorg/ace#5309] [Q&A]
hello

if i try next html in kitchen demo https://ace.c9.io/build/kitchen-sink.html

beautify ctrl + shift + b
it did not work
```
<meta name="copyright" content="The Owner"/>         />= problem
<meta name="copyright" content="The Owner">           >=working
but no allert in html for />
```

```
<!DOCTYPE html >
<html>
<head>
<meta http-equiv="content-type" content="text/html; charset=utf-8" />
<title>`substitute(Filename('', 'Page Title'), '^.', '\u&', '')`</title>

</head>
<body>

<iframe src="" frameborder="0"></iframe>

<select name="option select" id="">
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
</select>
</body>
</html>
```

i get 
![2023-09-10-180830_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/29a5fe82-210e-43d3-815e-9962d2e87750)

if i do

```
<!DOCTYPE html >
<html>
<head>

<title>`substitute(Filename('', 'Page Title'), '^.', '\u&', '')`</title>

</head>
<body>

<iframe src="" frameborder="0"></iframe>

<select name="option select" id="">
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
<option value="option select">opt select</option>
</select>
</body>
</html>
```

wthout meta it works
why no errors in first try

![2023-09-10-181113_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/5eba57a3-6a4c-4cc9-a89b-08250e0d21da)

if i try it in old ace test from me
it hangs on the meta copyright
but works if the meta is removed

![2023-09-10-182739_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/1f7c399c-07ca-4c78-b239-ac79c63a11e7)

![2023-09-10-182936_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/4ea27ef0-148a-4b30-9010-a8da99d5d96b)

---

https://plnkr.co/edit/Digj5Fw7D6hxaTU4

![2023-09-10-184744_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/dbeebaee-7e45-4632-b543-23244251e6fd)

https://plnkr.co/edit/Digj5Fw7D6hxaTU4

**Comment by ldijkman** [comment]
maybe because i use an older version
looks like in next link it works ok
https://mkslanc.github.io/ace-playground/#scrollbars
think it uses
```
<script src="https://www.unpkg.com/ace-builds@latest/src-noconflict/ace.js"></script>
```

![2023-09-10-210857_1920x1080_scrot](https://github.com/ajaxorg/ace/assets/45427770/9a38d724-43e0-445d-a97e-b040fcc9dd5f)

**Reply by whazor** [reply]
I think the mkslanc ace-playground uses [ace-linters](https://github.com/mkslanc/ace-linters) for formatting.

**Reply by ldijkman** [reply]
Thank You 
i will have a look 

https://mkslanc.github.io/ace-playground/#hello-world
https://github.com/mkslanc

greet Luberth
https://plnkr.co/edit/Digj5Fw7D6hxaTU4
